src/auth/authenticator/oicd/authenticator.go

```diff
@@ -0,0 +1,70 @@
+package oicd
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/consensys/quorum-key-manager/pkg/tls/certificate"
+	"github.com/consensys/quorum-key-manager/src/auth/authenticator"
+	"github.com/consensys/quorum-key-manager/src/auth/types"
+)
+
+const AuthMode = "JWT"
+
+type Authenticator struct {
+	jwtCheckers []*JWTChecker
+}
+
+var _ authenticator.Authenticator = Authenticator{}
+
+func NewAuthenticator(cfg *Config) (*Authenticator, error) {
+	if len(cfg.Certificates) == 0 {
+		return nil, nil
+	}
+
+	auth := &Authenticator{
+		jwtCheckers: []*JWTChecker{},
+	}
+
+	for _, cert := range cfg.Certificates {
+		keyPair, err := certificate.X509KeyPair([]byte(cert), nil)
+		if err != nil {
+			return nil, err
+		}
+
+		auth.jwtCheckers = append(auth.jwtCheckers, NewJWTChecker(keyPair.Leaf, cfg.Claims, false))
+	}
+
+	return auth, nil
+}
+
+func (a Authenticator) Authenticate(req *http.Request) (*types.UserInfo, error) {
+	// Extract Access Token from context
+	token, ok := extractToken("Bearer ", req.Header.Get("Authorization"))
+	if !ok {
+		return nil, nil
```

Comment:
Does this mean we don't fail when no Bearer token is passed? Shouldn't we return an error and ignore it on the parent?

Reply:
Yes, in case no token is passed Auth validator is skip and deletegate to next authenticator (TLS, API_KEY) to extract the user info. In case none of them are failed user will be obtain the `anonymous` role and whatever that role has permissions for